src/app/modules/payments/stripe.controller.ts

```diff
@@ -122,11 +122,27 @@ const _handleStripeEventUpdates: ControllerHandler<
     case 'charge.pending':
     case 'charge.refunded':
     case 'charge.succeeded': {
+      let paymentIdForEmail
+
       result = await getMetadataPaymentId(
         event.data.object.metadata,
-      ).asyncAndThen((paymentId) =>
-        StripeService.processStripeEvent(paymentId, event),
-      )
+      ).asyncAndThen((paymentId) => {
```

Comment:
Im thinking if there are ways to avoid having `let paymentIdForEmail` be declared

Reply:
Sorry for the super late response. I was thinking something like this, although there's a weird `okAsync` out of nowhere. Not sure if this is better either 🤣 

```ts
    case 'charge.succeeded': {
      result = await getMetadataPaymentId(
        event.data.object.metadata,
      ).asyncAndThen((paymentId) => {
        const processEventResult = StripeService.processStripeEvent(
          paymentId,
          event,
        ).andThen(() => {
          if (event.type === 'charge.succeeded') {
            return PaymentService.sendPaymentConfirmationEmailByPaymentId(
              paymentId,
            )
          }
          return okAsync(undefined)
        })

        return processEventResult
      })

```

Reply:
Thanks for the suggestion Ken! Have to do more wrangling to get Typescript to be happy with that suggestion :"( I'm going to leave it as it is for now, given the time urgency of this issue.